js/options/colorfulAudio.js

```diff
@@ -0,0 +1,101 @@
+/* global C2D */
+window.options.colorfulAudio = {
+  name: 'colorful audio',
+  state: {
+    selected: false
+  },
+  run: function () {
+    const div = document.createElement('div')
+    const btnColor = C2D.stroke
+    div.innerHTML = `
+      <style>
+        div{
+          display:flex;
+          align-items:center;
+          justify-content:center;
+          flex-direction:row;
+        }
+        button{
+          width:500px;
+          height:90px;
+          margin-right: 40px;
+          background-color:${btnColor};
+          color:white;
+          font-size:200%;
+          font-family:"Monaco", monospace;
+          border-radius:10%;
+        }
+      </style>
+
+      <audio id="inputAudio"></audio>
+
+      <div>
+        <button> Record Audio </button>
+        <audio id="savedAudio" controls></audio>
+      </div>
+    `
+    function colorfulSounds () {
+      const savedSound = document.getElementById('savedAudio')
+      const randomNum = Math.random()
+
+      savedSound.onplay = () => {
+        const pixels = C2D.getPixels()
+        const screenWidth = C2D.width
+        const portion = (C2D.height / 100)
+        let partialRow = 1
+        let accumulator = 1
+        let currentColor = 3
+        let time = savedSound.currentTime
+
+        if (time > 0 < 1) { time = 1 }
+
+        const soundMultiplier = (savedSound.volume *
+          time / savedSound.duration)
```

Comment:
so, if i understand this correctly, u're algorithm crates random colors (which it replaces all the non-transparent pixels on the screen with) && then it scales those random colors based on the current volume level (ie. the volume knob in the controls) && the length of the sound. why did u choose to use the volume control level instead of the actual differences in volumes within the file recording itself? it seems to me that the latter would produce a more interesting result no? for example, if u used ur `UserMedia.getAudioIntensity` instead of just the audio slider level

Reply:
When I tested and did more research into the Web Audio API, I found out that the analyzer node (which gives audio intensities in dB) only seems to work with live audio, so when I run the UserMedia.getAudioIntensity on the recorded audio, it returns an array of zeroes. Apparently, to fix this problem I would need to use another library like tone js. Since my initial idea was to use a user-generated audio file, I thought it would be interesting to shift gears a bit. However, I definitely see how this can be made more interesting! One idea I have is to create a while loop to run the code while the audio is being played and thus change the screen color in real time. For now, maybe the best thing to do is merge the repos so that others can use the UserMedia library, and I'll improve the colorfulAudio option later

Reply:
Should be better now!